File: src/main/java/olivermakesco/de/refmagic/material/WitheriteArmorMaterial.java
package olivermakesco.de.refmagic.material;

import net.minecraft.entity.EquipmentSlot;
import net.minecraft.item.ArmorMaterial;
import net.minecraft.item.ArmorMaterials;
import net.minecraft.recipe.Ingredient;
import net.minecraft.sound.SoundEvent;
import olivermakesco.de.refmagic.registry.RefinedMagicItems;

public class WitheriteArmorMaterial implements ArmorMaterial {
    public static WitheriteArmorMaterial instance = new WitheriteArmorMaterial();

    @Override
    public int getDurability(EquipmentSlot slot) {
        return ArmorMaterials.NETHERITE.getDurability(slot);
    }

    @Override
    public int getProtectionAmount(EquipmentSlot slot) {
        return ArmorMaterials.NETHERITE.getProtectionAmount(slot);
    }

    @Override
    public int getEnchantability() {
        return ArmorMaterials.NETHERITE.getEnchantability();
    }

    @Override
    public SoundEvent getEquipSound() {
        return ArmorMaterials.NETHERITE.getEquipSound();
    }

    @Override
    public Ingredient getRepairIngredient() {
        return Ingredient.ofItems(RefinedMagicItems.witheredHeart);
    }

    @Override
    public String getName() {
        return "witherite";
    }

    @Override
    public float getToughness() {
        return ArmorMaterials.NETHERITE.getToughness();
    }

    @Override
    public float getKnockbackResistance() {
        return ArmorMaterials.NETHERITE.getKnockbackResistance();
    }
}

File: src/main/kotlin/olivermakesco/de/refmagic/material/WitheriteArmorMaterial.kt
package olivermakesco.de.refmagic.material

import net.minecraft.entity.EquipmentSlot
import net.minecraft.item.ArmorMaterial
import net.minecraft.item.ArmorMaterials
import net.minecraft.recipe.Ingredient
import net.minecraft.sound.SoundEvent
import olivermakesco.de.refmagic.registry.RefinedMagicItems

class WitheriteArmorMaterial : ArmorMaterial {
    override fun getDurability(slot: EquipmentSlot): Int {
        return ArmorMaterials.NETHERITE.getDurability(slot)
    }

    override fun getProtectionAmount(slot: EquipmentSlot): Int {
        return ArmorMaterials.NETHERITE.getProtectionAmount(slot)
    }

    override fun getEnchantability(): Int {
        return ArmorMaterials.NETHERITE.enchantability
    }

    override fun getEquipSound(): SoundEvent {
        return ArmorMaterials.NETHERITE.equipSound
    }

    override fun getRepairIngredient(): Ingredient {
        return Ingredient.ofItems(RefinedMagicItems.witheredHeart)
    }

    override fun getName(): String {
        return "witherite"
    }

    override fun getToughness(): Float {
        return ArmorMaterials.NETHERITE.toughness
    }

    override fun getKnockbackResistance(): Float {
        return ArmorMaterials.NETHERITE.knockbackResistance
    }

    companion object {
        var instance = WitheriteArmorMaterial()
    }
}